pkg/kubecli/fake_kubevirt.go

```diff
@@ -0,0 +1,114 @@
+package kubecli
+
+import (
+	"errors"
+
+	k8smetav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/tools/clientcmd"
+
+	k8sapiv1 "k8s.io/api/core/v1"
+	k8sv1 "k8s.io/client-go/kubernetes/typed/core/v1"
+
+	v1 "kubevirt.io/kubevirt/pkg/api/v1"
+)
+
+// parameter to control behavior of fake objects
+var InvalidFakeClient bool
+var InvalidFakeResource bool
+var InvalidFakeService bool
+
+// holding a copy of the created service, so that the test code could verify
+var CurrentFakeService *k8sapiv1.Service
+
+// entry point for testing, could be used to override GetKubevirtClientFromClientConfig
+func GetFakeKubevirtClientFromClientConfig(cmdConfig clientcmd.ClientConfig) (KubevirtClient, error) {
```

Comment:
Why can't you use our mock client?

Reply:
because I want to more easily test all branches of my code, without running that via a mock http server.
doing that at the mock http server level would require much more complexity in the test code. note that other commands either dont have any unit-tests, or have unit tests that does not cover anything. 
this will allow adding good coverage to all commands

Reply:
The mock client does not involve a http server. Here an example on how it works:


```
		virtClient = kubecli.NewMockKubevirtClient(ctrl)
		vmInterface = kubecli.NewMockVMInterface(ctrl)
		virtClient.EXPECT().VM(k8sv1.NamespaceDefault).Return(vmInterface).AnyTimes()
                vmInterface.EXPECT().Update(myupdatedvm)
```

It uses gomock.

Reply:
this is based on a generated file pkg/kubecli/generated_mock_kubevirt.go
but if you look at my mock code at pkg/kubecli/fake_kubevirt.go I also need a mock for the ServiceInterface. how do i add that to the generated file?

Reply:
You can use the client-go fake client for k8s objects. See for instance here: https://github.com/kubevirt/kubevirt/blob/master/pkg/virt-controller/watch/node_test.go#L76.

Reply:
Let's not add a third technology for a fake client. I would prefer if you use the k8s fake client for k8s resources.

Reply:
will remove this, and use existing one